pkg/apis/triggers/v1alpha1/event_listener_types.go

```diff
@@ -39,6 +39,8 @@ type EventListenerTrigger struct {
 	Binding         EventListenerBinding  `json:"binding"`
 	Template        EventListenerTemplate `json:"template"`
 	Params          []pipelinev1.Param    `json:"params,omitempty"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
Agree with @dibbles 's comment here -- does supporting all the other fields in ObjectMeta make sense here? e.g. namespaces, generateName, selfLinks etc? If not, maybe a name of type string should suffice?

Also if we are going to use objectMeta, the Trigger type is starting to look a lot like its own CustomResource. In that case, the other fields like Binding, Template, Params etc should be within a Spec?

Reply:
Yea. Should we have Trigger CRD? Will change to just string `name` for now and create an issue regarding this so that we can discuss this properly.